chain/default_syncer.go

```diff
@@ -367,3 +370,21 @@ func (syncer *DefaultSyncer) HandleNewBlocks(ctx context.Context, blkCids []cid.
 	}
 	return nil
 }
+
+// IsReorg determines if a new tipset would cause a reorg if heavier than the
+// syncer's head.  The new tipset will cause a reorg if the syncer's head is
+// not one of its ancestors or a subset of the new tipset.
+func (syncer *DefaultSyncer) IsReorg(newAncestors []types.TipSet, new types.TipSet) bool {
```

Comment:
the tests for this would be so much simpler if the caller passed in what the function needed, instead forcing the function to do it. i see this pattern a lot, it makes for lots of coupling and makes functions hard to understand. this thing is much simpler if it is a free function that is passed two things: 1 a slice of tipsets representing the new chain including the new head and 2 the current head. think about how much cleaner the unit tests can be, no syncer required at all, just tipsets.

this is the same pattern i saw with the lookback stuff: making functions hard to understand so the caller can be cleaner. it should be the other way around.